internal/workflow.go

```diff
@@ -692,25 +690,24 @@ func (wc *workflowEnvironmentInterceptor) ExecuteChildWorkflow(ctx Context, chil
 		}
 	}, func(r WorkflowExecution, e error) {
 		if e == nil {
-			childWorkflowExecution = &r
+			if cancellable {
```

Comment:
What happens if initiation takes a long time?  Won't we still be stuck on cancellation and waiting for initiation to go through?

Reply:
added a comment explaining why we are waiting on initiation